3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,77 @@
+//Voice assistant
+let personName = "";
+let toDoList = [];
+
+function getReply(command) {
+  if (command.startsWith("Hello my name is")) {
+    personName = command.split(" ")[4];
+    console.log(`nice to meet you ${personName}`);
+    return;
+  }
+
+  if (command === "What is my name") {
+    if (personName === "") console.log("I dont have your name");
+    else {
+      console.log(personName);
+    }
+    return;
+  }
+
+  if (command.startsWith("Add") && command.endsWith("to my todo")) {
+    // I did this part with for loop but i found easier way
+    //let commandArr = command.split(" ");
+    // for (let i = 1; i < commandArr.length - 3; i++) {
+    //   if (i == commandArr.length - 4)
+    // toDo += commandArr[i];
+    //   else toDo += commandArr[i] + " "; }
+
+    let toDo = "";
+    toDo = command.substring(4, command.length - 11);
+    toDoList.push(toDo);
+    console.log(`${toDo} added to your todo`);
+  }
+  if (command.startsWith("Remove")) {
+    let toRemove = "";
+    toRemove = command.substring(7, command.length - 14);
+  }
+
+  if (command === "What is on my todo?") {
+    let myList = "";
+    for (let i = 0; i < myList.length; i++) {
+      toDoList += myList + "and";
+      if (i === myList.length - 1) toDoList += myList;
+      else {
+        toDoList += myList;
+      }
+      console.log((toDoList += myList + "and"));
+    }
+  }
+
+  if (command === "What day is it today?") {
+    let date = new Date();
+    console.log(`Today is ${date}`);
+  }
+  if (command.startsWith("What is")) {
+    let math = "";
+    math = command.substring(8, command.length);
+    console.log(eval(math));
```

Comment:
This is a clever choice that allows for quite complex math as well! However, it is highly dangerous to use this command, and I will recommend avoiding. As an example, I could ask `What is alert('You have been hacked!')` and it would show a message on the screen. This could be used by hackers to stir up trouble! 😱

I'd suggest simplifying the math you will allow and code something simple that handles the most basic operations on two numbers such that you can avoid calling `eval`.